sympy/core/containers.py

```diff
@@ -141,25 +141,27 @@ def index(self, value, start=None, stop=None):
         else:
             return self.args.index(value, start, stop)
 
-    def _eval_Eq(self, other):
-        from sympy.core.function import AppliedUndef
-        from sympy.core.logic import fuzzy_and, fuzzy_bool
-        from sympy.core.relational import Eq
 
-        if other.is_Symbol or isinstance(other, AppliedUndef):
-            return None
+converter[tuple] = lambda tup: Tuple(*tup)
+
 
-        if not isinstance(other, Tuple) or len(self) != len(other):
-            return S.false
+@dispatch(Tuple, Basic)
+def _eval_Eq(lhs, rhs):
+    from sympy.core.function import AppliedUndef
+    from sympy.core.logic import fuzzy_and, fuzzy_bool
+    from sympy.core.relational import Eq
 
-        r = fuzzy_and(fuzzy_bool(Eq(s, o)) for s, o in zip(self, other))
-        if r is True:
-            return S.true
-        elif r is False:
-            return S.false
+    if rhs.is_Symbol or isinstance(rhs, AppliedUndef):
+        return None
```

Comment:
Maybe it would make sense to do the type checking in dispatch by having one handler for `Symbol` and one for `AppliedUndef` and then another for `Tuple`.

Reply:
Is every object with is_Symbol guaranteed to be of type Symbol? I think there is one case, I can find out more details for you, but if I do find that there is class that does not subclass Symbol but has is_Symbol set to true should I edit to subclass Symbol

Reply:
I'm not sure. What is the example?

Reply:
nevermind, it did actually work.